app/assistify/migrations/server/startup/migrations.js

```diff
@@ -0,0 +1,121 @@
+import { Meteor } from 'meteor/meteor';
+
+import { Users, Messages, Rooms, Subscriptions } from '../../../../models/server';
+import { getUserAvatarURL } from '../../../../utils/lib/getUserAvatarURL';
+import { roomTypes } from '../../../../utils/server';
+
+/*
+Up to now, there's no "DB version" stored for assistify.
+Until we've got expensive of contradicting migrations, we'll just use this file to write functions running
+on startup which migrate data - ignoring the actual version
+ */
+
+Meteor.startup(() => {
+	const _guessNameFromUsername = function(username = '', email = '') {
+		return (username || email.replace(/@.*/, ''))
+			.replace(/\W/g, ' ')
+			.replace(/\s(.)/g, function($1) { return $1.toUpperCase(); })
+			.replace(/^(.)/, function($1) { return $1.toLowerCase(); })
+			.replace(/^\w/, function($1) { return $1.toUpperCase(); });
+	};
+
+	const usersWithoutName = Users.find({ name: null }).fetch();
+	usersWithoutName.forEach((user) => {
+		if (user.username || (user.emails && user.emails.length > 0)) {
+			Users.update({ _id: user._id }, { $set: { name: _guessNameFromUsername(user.username, user.emails[0].address) } });
+		}
+	});
+});
+
+
+/*
+ Migrate from Threads to Discussions.
+ - In Rooms Schema ParentRoomId changed to prid
+ - In Subscription schema ParentRoomId changed to prid
+ - Messages with type 'thread-welcome' will be removed.
+ - Messages with type 'create-thread' will be migrated to 'disscussion-created'.
+ - Author Information and Message link will be encapsulated inside attachments of message type 'disscussion-created'.
+ */
+
+Meteor.startup(() => {
+	console.log('Migrating from threads to Discussions');
```

Comment:
Threads to discussions are already migrated. We might not need this function anymore.